### ISSUE TITLE
Easier support for Beta version

### ISSUE THREAD

**Issue by Ipstenu** [Chassis/Chassis#838]
Right now, in order to test beta WP I download and overwrite the version with Chassis, which isn't a problem except when I want to update Chassis base code I have to clear it out, update, and re-download.

I feel like there should be a better way to flag and say that I want to use beta/RCs for testing pre-WP releases :)

**Comment by BronsonQuick** [comment]
Hey @Ipstenu! Great to hear from you and thanks for opening an issue 😁 

We do have the concept of [paths](https://docs.chassis.io/en/latest/config/?highlight=paths#paths) in Chassis for this kinda stuff. We added that so you can kind of "drop in" Chassis alongside an existing code base.

That being said, there's something better than that which might help you out. A while ago, Kadam, Peter Wilson and I had a crack at creating a [WP core dev](https://github.com/Chassis/core_dev) extension for Chassis a while ago to help with this kinda stuff. Full disclosure, I haven't actually run it in a while but if you wanna take it for a spin and let me know how you go then I'm happy to assist, enhance or improve it. I might actually pop it on my list of things to do on the weekend seeing it's been ages since I've used it! Let me know how you go, I'm always happy to help out! 💖

**Comment by Ipstenu** [comment]
Installing core-dev now to have a go with it 👍 It's relatively straight forward. Since I wasn't using paths (my chassis install was 'clean' as it were) I didn't even think about using it that way!

**Comment by BronsonQuick** [comment]
Awesome. Let me know if you have an issues, questions or feedback. I'm always happy to help!

**Comment by Ipstenu** [comment]
As soon as I use paths (for core-dev or even just installing beta in another folder), it edits wp-config.php, which now shows up as my git pull being modified. This (likely) won't cause issues, but it's disturbing. (I use https://ohmyz.sh/ as it lets me see Git status super easily as well as a bunch of other neato features, this means it's way more obvious to me than to others.)

Core Dev wants me to use NPM which is actually why I didn't want to use it (for dev'ing core itself that's fine, but I'm trying to test my website and plugins against betas, so it's really waaaay overkill). Not to mention NPM on Mac is constantly a jerk right now, what with "You don't have grunt sass!" errors (stupid mac....).

It makes me think the 'right' direction would be to not have wp-core in Chassis at all. Or be in an ignored if edited folder. Then I could define in config.local.yaml the wordpress version, and on provision, it would use wp-cli to install that?

**Comment by BronsonQuick** [comment]
Yeah I wasn't a big fan of the `wp-config.php` edit either. It was a hack we had to do because of a check `WP-CLI` does during an install.

Great feedback! Based on your feedback I'm thinking that maybe we could do a `beta-tester` extension for Chassis and figure out a good way to do this. I like how Westie's Beta Tester plugin handles this with a default WP install. I'll see if I can figure out a super simple way to do this. It could be as simple as a Chassis extension that uses the `paths` functionality and automagically installs the WordPress RCs into a folder that's mapped with paths.

**Comment by BronsonQuick** [comment]
Ryan and I partially regret using a subtree for the `wp` folder but at the time it made sense. I'm sure we can come up with a simpler solution for testing betas ✨
